composer/devices/device.py

```diff
@@ -90,6 +90,20 @@ def optimizer_to_device(self, optimizer: Optimizer) -> Optimizer:
                     state[k] = self.tensor_to_device(v)
         return optimizer
 
+    def decode_batch_to_dict(self, batch: T_Batch) -> T_Batch:
```

Comment:
why is this necessary?

Reply:
this method doesn't change the content or shape or anything, but without this method, the batch type is sometimes `'transformers.tokenization_utils_base.BatchEncoding'>`, which `get_num_samples_in_batch` and `get_num_tokens_in_batch` can't handle. this just converts it to a dict mapping.

Reply:
Can you elaborate and/or give an example transformation? I still don't follow

Reply:
synced offline

tldr `get_num_samples_in_batch` and `get_num_tokens_in_batch` throw NotImplemented Errors when they are called before a batch is loaded onto a gpu bc they don't recognize `'transformers.tokenization_utils_base.BatchEncoding'>`